### ISSUE TITLE
Make Volunteer Events Volunteerable

### ISSUE THREAD

**Issue by bethlakshmi** [bethlakshmi/gbe-divio-djangocms-python2.7#19]
Taking down notes from discussion in the car on the way to rehearsal, so @burlexpo can update later:

- All Volunteer events (and only those) for a given Conference will show up as options to volunteer for when a user is proposing to be a volunteer (ie, making a bid).
     - a volunteer can select and de-select any opportunities
     - volunteeer must have both email address and phone number in profile to select - (I should prevent them from removing phone/email if they are signed up as volunteers in a current conference).
- Volunteer events can be marked (by staff leads/volunteer coordinator) as one of two things:
    - requires approval - the staff lead must review the person volunteering and mark them as approved before the volunteer is actually assigned to work the opportunity.
    - doesn't require approval - as soon as the volunteer selects and submits the event, they are assigned.
- Assignment is:
     - event appears on volunteer's schedule
     - volunteer shows up in all scheduling reports 
     - counts toward event being full
- A volunteer does not count as "assigned" (and therefore the event remains open) if they are awaiting approval - so an infinite number of volunteers can be awaiting approval, even if there is only 1 open slot.
- whenever a volunteer adds/removes opportunity from their selections, the site will send email:
   - to the volunteer - confirming what just happened (assigned to event, removed from event, awaiting approval for working event).
   - to the staff lead - when it's a no-approval event, or de-selecting event - the email is just an FYI (no action needed), when it's application for an approval-event, the staff lead is prompted to review and approve/deny participation on the website.
- there will need to be a page that lets staff leads approve volunteers... 

- a user is blocked from causing themselves conflicts:
     - they are not allowed to volunteer for an event if they have already volunteered in an overlapping time slot, or they are teaching, or they are in tech rehearsal, or they are performing in show.

**Comment by bethlakshmi** [comment]
Questions:

-------
- I think the approval page should be a full list per staff area and/or per scope the user is allowed to see (ie, if a user is running say registration and merch, they see both in 1 screen?).  
    - So it shows:  every user's application for approval for every event w/in the lead's scope.
    - selection, however, is on an item-by-item basis (not bulk update).
    - what other context needs to be provided?

SAMPLE IDEA:

A table with the following columns:

Volunteer:  Elizabeth Kelly (ie, profile name) (link to full proposal)
Volunteer Event:  Registration Lead (link to full event details including current volunteers)
Time:  8AM-12PM
# Slots: 1 of 2 (1 is free, out of 2 total)


--------

Next Q:  Would we remove anything from the current volunteer info?

We have a times available/not available - do we still want them?
And interests...  
And other stuff....

**Comment by bethlakshmi** [comment]
OK Stopping.

**Comment by burlexpo** [comment]
> Would we remove anything from the current volunteer info?
Yes, everything.  I don't think we care what people's interests or availability are.  They're showing us them by signing up for a particular role at a particular time.

**Comment by burlexpo** [comment]
> Also - do we let volunteers create overlaps if they are "interested" in something?
> Should that just be a warnable condition?

Yes, they can create overlaps with "interests".  No, we don't need to warn them.

**Comment by bethlakshmi** [comment]
Got it.  Starting on this for offline work as I can...

**Comment by bethlakshmi** [comment]
ok, there's a lot here, so I am planning out how to do a series of tickets that optimizes us towards getting the volunteer recruitment part out FAST, and then follows up after with the management parts.  Here's my work breakdown, it'll probably evolve:

**1 - Ground work - make event schedule items have a approval-needed flag (DONE)**

This will be a marker on the scheduled case of an event (which means it's available to ANY event type (Volutneer, Show, Class, etc).  And it's unique to each schedule case.  So "Registration" for 2 hours at 8:30AM can need approval, while "Registration", for 2 hours with the same description, at 10:30AM *doesn't* need approval.

**2 - Set up volunteer signup:**

- replace the current volunteer bid with a signup for the volunteer events that have been set up.
- my simple approach will be a big old list, that is sortable and filterable, like Volunteer Assign (from the review volunteer list page) - but without windows and interests and such.  It will have:
    - title & a hover to show the description (or the description of it's staff area)
    - start date/time
    - end date/time 
    - "Area" = Staff area, show name, relevant interest tags - the idea is to give some strings for people to search for.

I'll use the table that I've used for reviewing lists all over the website.  There's the known bug around sorting by date and given just how many volunteer events we have - that's a real liability.  I'll spend a little time up front trying to find a fix here.  But I'm timeboxing that to an afternoon...

LATER - I strongly suspect we want to let people filter by date/time range. That's worth it's own ticket.

Also on this item - when people sign up, they get a confirmation alert on the screen.  If the event requires approval, they are told offer is pending.  For any approval not needed, we just thank them.

Also - they can go BACK to the signup page, and UNSIGNUP.  That also gives alert confirmation.  It also requires that I show user's current status on this page, it can't just be a blank slate.

DEBATABLE - if I can easily reuse code, I'll also send a notification to the relevant Staff leads and volunteer coordinator whenever a change occurs.  I won't bother with Volunteer Reviewers...  I may also skip sending a "approval needed" email, as that's real logic.

**3 - Make it easy (or at least sane) to configure approval on a volunteer event**

With item 1, approval for a given event can be configured in the admin tool.  That's really gruesome.  Betty can do it, Scratch do it but will really, really loose his mind if he has to do it a lot.  This step puts the approval configuration in the regular event management stuff in the Special -> Manage -> Schedule area.  Facets of this work:

- approval needed isn't configurable when copying - an event gets whatever the source of the copy had.
- in Staff Areas and shows, there are lists of volunteer events - make it configurable there.
- you can also configure each event individually by navigating to it directly - make it configurable there.
- we show a summary of # of volunteers needed vs. signed up on the main scheduling page - make some note there when there are pending volunteers that aren't approved yet.
- Only Volunteer Events will have this option.  While it's technically possible for ANY event to have approval configured on/off, logic is build so that only volunteer events have volunteers, so only volunteer events need approval to be configured.  The default for everything else is OFF.

**4 - Make it possible to approve people.**

- replace the review volunteer bids pages with the list of not-yet-approved volunteer offers.  
- privileged staff (volunteer coordinator and staff lead for volunteer opportunity) - can approve or reject these.  Approve makes them a committed volunteer, reject preserves the offer to volunteer but removes it from the to-be-approved list and does NOT make for a scheduled commitment.
- approval or rejection results in an email to the volunteer (unless they opted out w. email preferences) and approval results in an email to the staff lead.

ALSO - on the staff area/show management pages, we show a list of volunteer offers (ie volunteer bids).  My thought is to remove that (no one will be bidding) and replace it with the ability to approve offers for that staff area.  For a lead focused on only their one area, this is an easier way to see what your staffing picture looks like. 

At the moment, it's my intent to NOT show the volunteer coordinator all the booked volunteer offers - ie all the cases where a volunteer signed up for something that didn't need approval, as well as all the cases where approval has already been given.  They will see how many volunteers are currently assigned to an opportunity that needs approval (ie, 3 slots for stage kitten at the Main Event are available, 2 slots have been filled).

**5 - Clean up loose ends?** 

With this many moving parts, I'll have some.

**Comment by burlexpo** [comment]
**A thought on How To Display Volunteer Shifts**
Instead of wrestling with the "I can't deal with dates" bug, how about we display volunteer shifts like we do classes (and other BurlExpo events)? Instead of _Locations_ as column headers we use _Staff Area_, so you would go to Friday, April 17 and see 

![vol examples](https://user-images.githubusercontent.com/5369404/71646094-4c0ce080-2cb0-11ea-85f8-62bbc2caba37.jpg)

**Comment by bethlakshmi** [comment]
I'm trying to code this out and the calendar is subtly different than you're saying.  The existing calendar code is organized by start time - so events that start at 6PM are in the 6PM row.  If another event starts at the 7PM slot, then that event is on the 7PM row (that's the way classes work).  

Our volunteer events are fewer and more far between, but show stuff starts all at the same time.

Looking at GBE 2017 is an example:

https://burlesque-expo.us.aldryn.io/calendar/Volunteer?day=02-11-2017

This is Volunteer events on Saturday in 2017 (I skipped 2018 in case it was weird).

I mapped this page out into a column per track format and came up with this:

![Screen Shot 2020-01-03 at 8 29 03 PM](https://user-images.githubusercontent.com/4542707/71757914-aa2ff400-2e67-11ea-80cc-f3c6ec2371e6.png)

That's a LOT of white space.  The kicker is really all the one offs - our drop ins each have a volunteer option.  And there's many gaps because we only show a option during it's start hour, we don't fill it like a grid (look at registration  as an example).   

Proposal - it would be easier - pretty easy actually - to use the same format we have, but let people sign up by selecting a cell just like we do for "interested" - and then do the approval needed logic that way.

That would save be the complexity of trying to sort out columns.

It turns out, I also don't sort out columns on classes anymore - they are all just in the given hour.

**Comment by burlexpo** [comment]
1.  As I recall, we did the display of start time only to avoid situations of having display issues in the cells where events overlapped.
2. The Drop-in classes are a weird, special case. Creating them as Volunteer Slots was the best way I could handle them (they're not really classes and they're not always taught by the same person).  I guess we can have them displayed along with other Volunteer shifts; we'll just need to make them all Approval Needed.
3. 

> Proposal - it would be easier - pretty easy actually - to use the same format we have, but let people sign up by selecting a cell just like we do for "interested" - and then do the approval needed logic that way.

Since this is exactly what I suggested, I feel like you might be saying something else -- or are you just confirming we can do it this way?

**Comment by bethlakshmi** [comment]
OK... let's talk.  I've got two prototypes worth looking at posted on the test server.  To simulate our typical data load, I cloned our GBE 2017 staff areas over to this year.  I didn't port drop in classes, figuring we'd talk about them.  NOTE - this is only test data. 

I also took the liberty of volunteering you for 3 events, so you could see what that looks like:
 - you volunteered for "Art Show Setup" which is an approval-needed event, and so you are a pending volunteer
 - you also volunteered for a RR - Soundboard shift and a Security shift, and they don't require approval

#1 - the first way I tried:

https://burlesque-expo-stage.us.aldryn.io/volunteer/signup

- shows only events that have volunteer openings (not at max volunteers) OR events that you are signed up for as a volunteer (even if they are full)
- shows colors if you are a volunteer already (green) or pending approval (blue)
- the "i" icon appears if there is an event description, or if there is a staff area and the staff area has a description (event description takes precedence).  
- the time sorting bug is still a thing
- filterable by any string you see here... "Sound Board", "Rhinestone", "Registration", "04/17/2020" all work the way I think you'd want them to.  Limitation - you have to know to use our date format.  17/04/2020 won't get you anywhere, this is a string search.

2 - a calendar based approach that is strikingly similar to "interested" behavior:

https://burlesque-expo-stage.us.aldryn.io/calendar/Volunteer

I chose a little hand, since it looked volunteer-y

- volunteers can join/leave volunteer events, so can pending volunteers - one by one
- if your commitment to the event is NOT as a volunteer (or pending volunteer), then you are blocked from volunteering or un-volunteering - and told to contact the coordinator.  (ie, teachers, staff leads, etc).
- if you are a volunteer, teacher, moderator, panelist, or performer for this event, it is displayed as a pale pink (same as what we do with "interest")
- if your offer of volunteering is "pending" - the event is a pale blue
- further details are available by clicking on the event title 
- this shows 1 day at a time, with the option to scroll horizontally through the days.
- no filters
- NOTE - actually volunteering/unvolunteering hasn't yet been implemented.  If you click on the hand you'll get an error message.  Haven't finished that yet.  The #1 option actually works.

Questions:
- (regarding #2) - I did this as part of the Volunteer Calendar - is that OK?  Should it be a totally separate thing?  I can't think of what else we use this display for... so it seemed fair to add this to it...
- (regarding #2) - Would it be better to an "i" for info - like I did on the table - it saves people from lots of clicking and loading and isn't that hard.
- (regarding #2) - this approach could be extended to the event list/detail browsing - for "interested" the user can be interested almost anywhere - on the detail pages for specific event descriptions, on any lists of events, etc.  I could do the same thing here, as I'm mimicking that pattern...  it's work... but not insane work.  I'd advocate NOT doing it first-thing, but putting it  as a new item on my to-do list above if we want it.
- either one - should I do something to show events that will require approval BEFORE the user selects them?  Right now, the display shows blue if the user volunteered, and the volunteer offer is pending.  But we actually know that the event WILL require approval - so I could show it somehow?

For either of these, the alerting and email notifications aren't 100% what I have mind.  That's about the limit of what I prototyped before pausing to get input.

**Comment by bethlakshmi** [comment]
OK took notes from yesterday... here's attempt 3:

https://burlesque-expo-stage.us.aldryn.io/scheduling/volunteer_signup/?day=04-18-2020
(this is Saturday - so the busiest day)

- has the "shadow events" - but not the anchor links & icons
- has a popover (actually a modal) with details on every event - used it for details, staff area info, and a link to the parent event
- haven't figured out an marking for approval needed

**Comment by burlexpo** [comment]
The shadow events don't line up with the initial event.  I know changing the size of the screen will impact on this anyway, but right now because they don't line up, they don't solve the problem we're trying to fix, they're just adding ore confusion

Clicking on one of the shadow events threw an error --

Request Method: | GET
-- | --
https://burlesque-expo-stage.us.aldryn.io/scheduling/volunteer_signup/1014/
cms.views.details

**Comment by burlexpo** [comment]
Iconography thought --
I understand that the little hand is like "raising your hand" to volunteer but it also comes across as "stop".

How about we use a checkmark?

- green checkmark = open shift, can be volunteered for
- hollow checkmark (outline) = restricted shift, requires approval
- blue checkmark = you volunteered, it's been assigned to you.

I don't mean using simplelineicon; I mean using option-v (**√**)

**Comment by burlexpo** [comment]
Checkboxes.  I went with yellow for "needs approval" because of the "yellow=caution".  If you hate it, I can change the color.  They're GIFs, so should have transparent backgrounds.  They are 64x64px.  There are 5:
![Needs Approval - Gray Check](https://user-images.githubusercontent.com/5369404/71842228-28baaa80-308f-11ea-885a-eb9978acc7ad.gif)
![Needs Approval - Green Check](https://user-images.githubusercontent.com/5369404/71842229-28baaa80-308f-11ea-8af6-1bd671efff29.gif)
![Needs Approval - No Check](https://user-images.githubusercontent.com/5369404/71842230-28baaa80-308f-11ea-8fe7-e59c706e8152.gif)
![No Approval Needed - Green Check](https://user-images.githubusercontent.com/5369404/71842231-28baaa80-308f-11ea-9546-b74065b7c34c.gif)
![No Approval Needed - No Check](https://user-images.githubusercontent.com/5369404/71842233-28baaa80-308f-11ea-95ca-8cfa0165d6d3.gif)






1. No approval Needed - No Check (plain white checkbox; Volunteer shift does not need approval; user has not volunteered)
2. No approval Needed - Green Check (white checkbox with green check; Volunteer shift does not need approval; user has volunteered; approved automatically)
3. Needs Approval - No Check (plain yellow checkbox; Volunteer shift needs approval; user has not volunteered)
4. Needs Approval - Gray Check (plain yellow checkbox; Volunteer shift needs approval; user has volunteered; pending admin action)
5. Needs Approval - Green Check (plain yellow checkbox; Volunteer shift needs approval; user has volunteered; admin has approved)

**Comment by bethlakshmi** [comment]
Got it.  For the record, your "yellow" checkbox is green.  Very vibrant, lime green.  Which means that (to me) it's hard to see that the grey checkbox is grey.  I think the general look works, so I'll get the gifs coded into the site, and then we can replace the files as we like.

Also - they are huge, so I will force scale them down to a typical icon size (typical size ~16X16 px)

**Comment by bethlakshmi** [comment]
https://whentohelp.com/cgi-bin/w2h.dll/emptradeboard?SID=11991582214250&lmi=&Date=06/29/2020

I THINK THIS IS IT

I was going to give you other options, but I think this is what you want.

- because this is 1 row per opportunity, they can provide an infinite # of opportunities 
- they sorted by start time so as you go down, the opportunities get later in time
- they did a table (grumble, but this is really the only way) - and 1 column is 1 unit of time.  That lets them squash together columns 
- the mobile version (of course I loaded it on my phone), isn't wonderful, but it doesn't make me want to kill myself.  It's useable if not lovely.  They can't really do displays of less than 560px wide without being a bit ugly, but that's OK.
- clicking on a box opens a detail modal with signup options. 
- the characters on the row are VERY limited - but they have wrap around which makes a thicker box

We could do basically this. I would advocate:
- Put icons at the start of the line
- show a time grid that starts at 6AM, and ends at 4AM (just in case)
- time grid is 1 column per quarter hour, with dark lines on the hour (that's exactly what they do here)
- clicking on event brings up modal, modal has what our modal currently has plus:
     - the ability to signup/drop event 

This would mean you *can't* signup directly on the calendar.  With bars this narrow, that feels right to me - we don't want people accidentally withdrawing/adding events!

**Comment by burlexpo** [comment]
So... what do we do with this? Embed it in our site? Can we still communicate with volunteers with automatically triggered events and messages? Can we still use our internal email system to contact volunteers?

**Comment by bethlakshmi** [comment]
I know I said I'd just proceed... but since I hit a stopping place for the night:

https://burlesque-expo-stage.us.aldryn.io/scheduling/volunteer_signup/?day=04-17-2020

- the modal is there, but not complete.  I know it has no signup/unsignup links.  Also,  I plan to use the icons here, too.
- I can't get the small size grid to be evenly spaced  (like in mobile sized) - maybe I'll think of something but for now I'm out of ideas.
- I am debating updating the underlying structure to pull the left and right side margins on small-sized screens specifically for this page.  I can use every scrap of real estate.
- there's a bug on Sunday that I'm aware of - Strike runs to midnight...   I have ideas, but am too tired to work through them.

Then I have to get the action of signing up to work.

**Comment by bethlakshmi** [comment]
Taking a pause for tonight... 
- volunteer signup actually works
- on screen messaging in good shape - feedback alerts to user when they do something to change state (volunteer/un-volunteer) & instructions at top of list and for approval_needed events
- removed old volunteer list in table format.

TO DO:
- email to volunteer, volunteer coordinator, and staff leads every time a volunteer volunteers/un-volunteers
- make pending volunteer NOT show up on a person's schedule or as a conflict 
- do conflict checking & alert when a user signs up (either pending or volunteer direct) as an on-screen alert
- verify that these are final icons - the green checkboxes are a little weird, I think we really want yellow.
- consider visual issues (comment before this)
- change link on home page to this, and change verbiage on home page.
- set redirect for old volunteer bid to go here

**Comment by burlexpo** [comment]
Three new icons
![Needs Approval - Gray Check](https://user-images.githubusercontent.com/5369404/72122182-0f358f00-332b-11ea-8a69-41b54df8cd45.gif)
![Needs Approval - Green Check](https://user-images.githubusercontent.com/5369404/72122183-0f358f00-332b-11ea-9620-0ce4c387ebe7.gif)
![Needs Approval - No Check](https://user-images.githubusercontent.com/5369404/72122184-0f358f00-332b-11ea-8803-89cc98cefc4c.gif)

**Comment by bethlakshmi** [comment]
AWESOME!!!!  Probably getting 'em in tonight!!!

**Comment by bethlakshmi** [comment]
FYI - all changes are up to date on test server:

https://burlesque-expo-stage.us.aldryn.io/gbe
- has the new look for volunteers to signup (a button near their schedule)
- the signup calendar is still in test, but should be fully functional
- emails wil show up in /admin in the Post Office area.  The test server goes to great lengths NOT to mail the user, so the email MAY go to me... not you... but a record of it will be in admin, regardless.

- the template for staff emails doesn't show up in the Special -> Manage Template page yet, but it's in Post Office (/admin)
- the instructions at the top and in the note for pending-admission events is a UserMessage, editable in admin. 
- same for all the warnings/alerts that show up on the user's page when they do something
- you shouldn't need to change static pages in Django CMS - the old link (volunteer/create) should now redirect to the new stuff.

Not too much more code is left but there is a LOT of testing to do here.  I took the old way apart a bit, and so I have a lot of tests to update.

**Comment by bethlakshmi** [comment]
Pull request for the volunteer side of this is ready.  

Moving on to management side.  First things on next effort:
- make sure that the email template for the email to the volunteer coordinator/staff leads is editable in our normal way.
- make it so that events that need approval can be marked in our schedule management code (right now, it has to be done though /admin, and that's painful)

**Comment by bethlakshmi** [comment]
@burlexpo - Working on the next step - I think we don't need the "volunteer type" anymore.  This was a fixed list, that you set for each volunteer event, and then the volunteer can also set it in their bid as an interest - so it helps with match making.

Volunteers now pick based on event title & description, so I figure I can remove it.  That means it won't show up here:

https://burlesque-expo.us.aldryn.io/scheduling/manage
("Volunteer Type" filter)

Or here:

https://burlesque-expo.us.aldryn.io/scheduling/show_edit/GBE2020/4605? 
Under "Manage Volunteer Opportunities"

The latter is what I want to remove most - because it takes up a bunch of space on the table, which is already crowded and getting bigger with a checkmark for "Approval".  You'd think a checkmark wouldn't take up much space, but it actually makes a big difference.

I won't pull the old data just yet, but I think I need to cleanup.

**Comment by bethlakshmi** [comment]
OK... now that step 1 & 2 are live!!

The test server has my work on making it easy to manage setting whether an event needs approval.  Things to play with:

**Special Menu -> Schedule -> All Events**
- edit any staff area 
- edit any show
....

And under the list of volunteer events you'll see that "volunteer type" is gone, but "Approve" is there.  Click any checkbox on and off and press the pencil button to save the row.

Duplicate also works - the Approve setting will persist from copied event to target.

**Same starting place -- edit any Volunteer Event**
- there's an approve button near the bottom same as above.  Works as it should.

**Same starting place -- Create button**

If you pick a volunteer opportunity to create, and choose to make an opportunity w/out a staff area or parent event - you should see the same screen as the Edit Volunteer Event.  If you pick a staff area or show, it will appear the same way as editing a staff area or show does (in a row of events)

The last thing I need to check/fix is whether or not copy works.   Obviously copying a volunteer event should result in the same state approve/need to approve.

**Comment by bethlakshmi** [comment]
Everything from the comment above is now merged and updated on production.  No need to use the /admin any more, although I don't have a search set up yet for "needs approval" events in the pages I built.

Now I'm posting the first part of step 4 - approve pending.  Go to Special -> Review -> Volunteers (same as before) and now you'll see the list of Pending, Waitlisted, and Rejected volunteers.  Any of them can be approved, waitlisted or rejected.  When you do that:
- it gives an alert on screen for what you did
- it gives an alert on screen for any conflicts IFF the case is that you set something to "Volunteer" - Rejected & Waitlisted don't count on your schedule.
- it gives an email to the Coordinator and the Staff Lead of the event (same as if the volunteer had signed up)
- it gives a schedule update email to the volunteer.

I want to:
- tune the emails - to clarify the types of state change better - waitlisted/rejected should not be a "schedule update" but something else....
- double check the volunteer signup display and make sure that:
     - wait listed volunteers show up as basically pending
     - rejected volunteers .... don't get the event on the list anymore?

And I need to write tests.

That will likely be the stopping place for this pull request.

After that, I think it's useful to replace the area in the volunteer event edit that lets a staff lead take a bidding volunteer and assign them, to taking a pending/waitlisted volunteer and assign them.  But that's a separate piece of work, so deserves a separate ticket.

**Comment by bethlakshmi** [comment]
Approve of pending volunteers as a review list is ready as a pull request.  Will review and push to site over the weekend.
Starting clean up and dead code removal.  - first target is all the Volunteer Bid assignment stuff.  Not review list, review, or edit.

The approval code is totally up to date and as it will be published on the test server, so it's available to play with.  Go to Special -> Review -> Volunteers and there it is.  It has a link to the old review page for old conferences, as well, so the old data is still available.

**Comment by bethlakshmi** [comment]
Review and approve pending volunteers is installed on the live site!!

I have a pull request pending on a bunch of cleanup code.
And I want to finish up with sorting out the event management list that we mentioned.

The site should now be 100% functional with the new volunteer signup scheme.

**Comment by bethlakshmi** [comment]
OK, I'm very close to finishing the 4th part of this - making it easy to see and manage the volunteers in all of their states in the volunteer opportunity schedule management pages.  

So... I'm thinking out what we need for reporting.  We currently have some decent (if I do say so myself) reports hidden away here:

https://burlesque-expo.us.aldryn.io/reports/staff_area

I think they need the following:

1 - put this very link as a menu item under "Reports" in the special menu, to increase it's availability.

2 - put a link to the menu items in the scheduling area, so a person working on a show, or staff area can quickly see what staff is already assigned... I propose an easy way - put a link to the follow on page, ie:

- https://burlesque-expo.us.aldryn.io/reports/staff_area/19?area=Staff
- https://burlesque-expo.us.aldryn.io/reports/staff_area/810?area=Show

(examples from GBE 2018) 
and put them as a link on the scheduler pages, such as:
- https://burlesque-expo.us.aldryn.io/scheduling/manage - with an icon next to the staff area or show
- https://burlesque-expo.us.aldryn.io/scheduling/show_edit/GBE2020/4605?  - in the "Manage Volunteer Opportunities" area

So that the user can see what volunteers are signed up.

** I don't want to make the data appear ON this page - not only is it visual overload... but loading these reports is still a bit slow and I don't want to make the user frustrated waiting for the page to load.  I can review loading this data in a less horrible way, but that is a BIG refactor and deserves it's own priority **

3 - refactor the report so that we are showing the same visual organizaiton but NOT showing users that are Rejected. 

QUESTIONS:
- on the report, should we show pending or waitlisted users?  my instinct is "no" - stick to those that are signed up for sure.  
- do i want to show the role - I could show all except rejected (for example) and include whether they are Voluntered, Pending, Waitlisted, etc?

**Comment by bethlakshmi** [comment]
4th update is live - all volunteer schduling makes note of the new feature!

**Comment by bethlakshmi** [comment]
Ahhh..... I think I'm done!!!

At least any new work deserves a new ticket, as the spirit my to-do list above is now DONE!

Just introduced the new reporting code on the live server.
And I deleted a fair amount of the old code.

I'll open up a later ticket for further cleanup.  But I recommend I do it after this expo.
